sycl/test-e2e/Matrix/Inputs/joint_matrix_bfloat16_impl.hpp

```diff
@@ -83,28 +94,37 @@ void test() {
   static constexpr size_t MATRIX_N = TN * 2;
   static constexpr size_t MATRIX_K = TK * 2;
   T A[MATRIX_M][MATRIX_K];
-  T B[MATRIX_K / 2][MATRIX_N * 2];
-  TResult C[MATRIX_M][MATRIX_N];
+  T B[MATRIX_K][MATRIX_N];
+
+  TAcc C[MATRIX_M][MATRIX_N];
   TResult D[MATRIX_M][MATRIX_N];
+  TResult DRef[MATRIX_M][MATRIX_N];
+
+  if (std::is_same_v<TAcc, bfloat16> || std::is_same_v<TResult, bfloat16>) {
+    matrix_rand<T>(MATRIX_M, MATRIX_K, (T *)A, T(1));
+    matrix_rand<T>(MATRIX_K, MATRIX_N, (T *)B, T(1));
+  } else {
+    matrix_fill(MATRIX_M, MATRIX_K, (T *)A,
+                [](int i, int j) { return (i == j) ? T(1) : T(0); });
+    matrix_fill(MATRIX_K, MATRIX_N, (T *)B, [](int i, int j) { return i; });
+  }
 
-  matrix_fill(MATRIX_M, MATRIX_K, (T *)A,
-              [](int i, int j) { return T(1) * (i + j); });
-  matrix_fill(MATRIX_K / 2, MATRIX_N * 2, (T *)B,
-              [](int i, int j) { return T(2) * i + T(3) * j; });
-  matrix_fill(MATRIX_M, MATRIX_N, (TResult *)C, TResult(1));
-  matrix_fill(MATRIX_M, MATRIX_N, (TResult *)D, TResult(1));
+  matrix_fill(MATRIX_M, MATRIX_N, (TAcc *)C, TAcc(0));
+  matrix_fill(MATRIX_M, MATRIX_N, (TResult *)D, TResult(0));
+  matrix_fill(MATRIX_M, MATRIX_N, (TResult *)DRef, TResult(0));
```

Comment:
Previously we initialized with `1`. I think we should keep it that way. It adds some additional testing for accumulation input.